crates/viewer/re_renderer/shader/world_grid.wgsl

```diff
@@ -0,0 +1,154 @@
+    switch (config.orientation) {
+        case ORIENTATION_XZ: {
+            plane_position += frame.camera_position.xz;
+            world_position = vec3f(plane_position.x, 0.0, plane_position.y);
+        }
+        case ORIENTATION_YZ: {
+            plane_position += frame.camera_position.yz;
+            world_position = vec3f(0.0, plane_position.x, plane_position.y);
+        }
+        case ORIENTATION_XY: {
+            plane_position += frame.camera_position.xy;
+            world_position = vec3f(plane_position.x, plane_position.y, 0.0);
+        }
+        default: {
+            world_position = vec3f(0.0);
+        }
+    }
+
+    out.position = frame.projection_from_world * vec4f(world_position, 1.0);
+    out.scaled_world_plane_position = plane_position / config.spacing;
+
+    return out;
+}
+
+// Like smoothstep, but linear.
+// Used for antialiasing: smoothstep works as well but is subtly worse.
+fn linearstep(edge0: f32, edge1: f32, x: f32) -> f32 {
+    return saturate((x - edge0) / (edge1 - edge0));
+}
+fn linearstep2(edge0: vec2f, edge1: vec2f, x: vec2f) -> vec2f {
+    return saturate((x - edge0) / (edge1 - edge0));
+}
+
+// Distance to a grid line in x and y ranging from 0 to 1.
+fn calc_distance_to_grid_line(scaled_world_plane_position: vec2f) -> vec2f {
+    return 1.0 - abs(fract(scaled_world_plane_position) * 2.0 - 1.0);
+}
+
+@fragment
+fn main_fs(in: VertexOutput) -> @location(0) vec4f {
+    // Most basics are very well explained by Ben Golus here: https://bgolus.medium.com/the-best-darn-grid-shader-yet-727f9278b9d8
+    // We're not actually implementing the "pristine grid shader" which is a grid with world space thickness,
+    // but rather the pixel space grid, which is a lot simpler, but happens to be also described very well in this article.
+
+    // Distance to a grid line in x and y ranging from 0 to 1.
+    let distance_to_grid_line = calc_distance_to_grid_line(in.scaled_world_plane_position);
+
+    // Figure out the how wide the lines are in this "draw space".
+    let plane_unit_pixel_derivative = fwidthFine(in.scaled_world_plane_position);
+    let line_anti_alias = plane_unit_pixel_derivative;
+    let width_in_pixels = 1.0;//config.thickness_ui * frame.pixels_from_point;
+    let width_in_grid_units = width_in_pixels * plane_unit_pixel_derivative;
+    var intensity_regular = linearstep2(width_in_grid_units + line_anti_alias, width_in_grid_units - line_anti_alias, distance_to_grid_line);
+
+    // Fade lines that get too close to each other.
+    // Once the number of pixels per line (== from one line to the next) is below a threshold fade them out.
+    //
+    // Note that `1/plane_unit_pixel_derivative` would give us more literal pixels per line,
+    // but we actually want to know how dense the lines get here so we use `1/width_in_grid_units` instead,
+    // such that a line density of 1.0 means roughly "100% lines" and 10.0 means "Every 10 pixels there is a lines".
```

Comment:
This seems inverted to me. A line density of 0.1 should mean a line every 10 pixels (or ui units).
The inversion of density is something like "screen space line spacing"